src/com/MPISinglePortsCommunication.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#ifndef PRECICE_NO_MPI
+
+#include <vector>
+#include "MPICommunication.hpp"
+#include "logging/Logger.hpp"
+
+namespace precice
+{
+namespace com
+{
+/**
+ * @brief Provides connection methods based on MPI ports (part of MPI 2.0).
+ *
+ * The two participants to be connected can be run in two process groups started
+ * up individually, i.e. not within the same process group.
+ */
+class MPISinglePortsCommunication : public MPICommunication
+{
+public:
+  explicit MPISinglePortsCommunication(std::string const &addressDirectory = ".");
+
+  virtual ~MPISinglePortsCommunication();
+
+  virtual size_t getRemoteCommunicatorSize() override;
+
+  virtual void acceptConnection(std::string const &acceptorName,
+                                std::string const &requesterName,
+                                int                acceptorRank) override;
+
+  virtual void acceptConnectionAsServer(std::string const &acceptorName,
+                                        std::string const &requesterName,
+                                        int                acceptorRank,
+                                        int                requesterCommunicatorSize) override;
+  
+  virtual void requestConnection(std::string const &acceptorName,
+                                 std::string const &requesterName,
+                                 int                requesterRank,
+                                 int                requesterCommunicatorSize) override;
+
+  virtual void requestConnectionAsClient(std::string      const &acceptorName,
+                                         std::string      const &requesterName,
+                                         std::set<int>    const &acceptorRanks,
+                                         int                     requesterRank) override;
+
+  virtual void closeConnection() override;
+
+private:
+  virtual MPI_Comm &communicator(int rank) override;
+
+  virtual int rank(int rank) override;
+
+  logging::Logger _log{"com::MPISinglePortsCommunication"};
+
+  std::string _addressDirectory;
+
+  /// Remote rank -> communicator map
+  std::map<int, MPI_Comm> _communicators;
```

Comment:
Why do we still need a map here? I thought, we would end up with one communicator, but apparently I misunderstood sth

Reply:
`acceptConnection` / `requestConnection` still uses one comm, created on `COMM_SELF`.

``acceptConnectionAsServer` / `requestConnectionAsClient` just uses `communicator[0]`, so it actually creates just one comm.

The reason of that is that the first variant is called in various different ways, mostly just one the accepting/receiving rank. In order to just use communicator, accept/request must be called on the entire global communicator. This would require quite some rework of the calling code. The gains would also be minuscule, because accept/request is mainly (only?) used for 1:1 connection anyways.

The latter, in contrast, is only called from `PointToPointCommunication`, due to that P2P was also heavily modified. But this connection method is the most important, because it does the m2n heavy lifiting.

If we agree, that `acceptConnection` / `requestConnection` just does 1:1 connection, we can rewrite and simplifiy the code.

Reply:
Currently, `accept/requestConnection` still does 1:N communications for the old server approach. I guess however that these occurrences could be replaced by `acceptConnectionAsServer` / `requestConnectionAsClient`. Anyway, could you simply please add a small comment in line 57 above to explain what you just wrote here.